src/main/java/uk/gov/pay/ledger/transaction/entity/TransactionEntity.java

```diff
@@ -228,6 +229,16 @@ public String getGatewayPayoutId() {
         return gatewayPayoutId;
     }
 
+    public void setFieldsFromOriginalPayment(TransactionEntity paymentTransaction) {
```

Comment:
Reading this, it's a bit hard to understand the semantic difference between `getFieldsFromOriginalPayment` and `setFieldsFromOriginalPayment`. 

`getFieldsFromOriginalPayment` is used to then update the refund event digest payload and `setFieldsFromOriginalPayment` is used separately to update the entity. 

Having the logic in both places means to follow this I have to understand the current method (combining event payloads for both resources) and the proposed method (setting entities).

If this could just read:
```java
mayBePaymentEventDigest.ifPresent(paymentEventDigest -> {
            TransactionEntity paymentTransactionEntity = transactionEntityFactory.create(paymentEventDigest);
            refundTransactionEntity.setFieldsFromOriginalPayment(paymentTransactionEntity);
        });
```

It would seem as clear as the original method, the "set" method would then be responsible for setting the `card_brand_label`, `expiry_date` and `wallet_type`. 

If that's too much work then potentially some more specific naming could make it clearer.

Reply:
Agree naming could be better. Updated to make the names clearer. I was aiming to have simple setters on entity or else it could bloat with responsibilities. If it is something we can do it in more simple way happy to do it in a separate PR.

Reply:
Naming changes look good, I think we're okay on "responsibilities" as it's one responsibility is to set fields based on a parent transaction, the only thing I can see right now is that it only does a few of the fields and not the ones that go inside transaction details (which it isn't immediately clear why).

 I agree it seems like something that would be a separate PR.